vcd/resource_vcd_network_direct.go

```diff
@@ -0,0 +1,115 @@
+package vcd
+
+import (
+	"github.com/vmware/go-vcloud-director/util"
+	"log"
+
+	"fmt"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/go-vcloud-director/govcd"
+	"github.com/vmware/go-vcloud-director/types/v56"
+)
+
+func resourceVcdNetworkDirect() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdNetworkDirectCreate,
+		Read:   resourceVcdNetworkRead,
+		Delete: resourceVcdNetworkDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"org": {
+				Type:     schema.TypeString,
+				Required: false,
+				Optional: true,
+				ForceNew: true,
+			},
+			"vdc": {
+				Type:     schema.TypeString,
+				Required: false,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"external_network": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"href": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"shared": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceVcdNetworkDirectCreate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	log.Printf("[TRACE] CLIENT: %#v", vcdClient)
+	vcdClient.Mutex.Lock()
+	defer vcdClient.Mutex.Unlock()
+
+	_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrgAndVdc, err)
+	}
+
+	extNetName, ok := d.GetOk("external_network")
+	externalNetworkName := ""
+	var externalNetwork types.ExternalNetworkReference
+	if ok {
+		externalNetworkName = extNetName.(string)
+		network, err := govcd.GetExternalNetworkByName(vcdClient.VCDClient, externalNetworkName)
+		if err == nil {
+			externalNetwork = *network
+		} else {
+			return fmt.Errorf("unable to find external network %s (%s)", externalNetworkName, err)
+		}
+	} // If no external network is provided, Terraform should fail the resource
+
+	orgVDCNetwork := &types.OrgVDCNetwork{
+		Xmlns: "http://www.vmware.com/vcloud/v1.5",
+		Name:  d.Get("name").(string),
+		Configuration: &types.NetworkConfiguration{
+			ParentNetwork: &types.Reference{
+				HREF: externalNetwork.HREF,
+				Type: externalNetwork.Type,
+				Name: externalNetwork.Name,
+			},
+			FenceMode:                 "bridged",
+			BackwardCompatibilityMode: true,
+		},
+		IsShared: d.Get("shared").(bool),
+	}
+
+	util.Logger.Printf("[INFO] NETWORK: %#v", orgVDCNetwork)
```

Comment:
This will look like random messages in info logs with strange information about what happend. "NETWORK:". Another question do we really want this in info level.

Reply:
I will remove it.